antarest/study/storage/rawstudy/model/filesystem/lazy_node.py

```diff
@@ -121,14 +142,33 @@ def save(self, data: Union[str, bytes, S], url: Optional[List[str]] = None) -> N
                 self.config.path.unlink()
             return None
 
-        self.dump(cast(S, data), url)
+        self.dump(t.cast(S, data), url)
         if self.get_link_path().exists():
             self.get_link_path().unlink()
         return None
 
+    def move_file_to(
```

Comment:
I don't really like the fact that this method does the move and can also copy files. I think you should separate things between 2 methods, one for copy and another for moving.